konrad/core.py

```diff
@@ -264,7 +274,22 @@ def run(self):
                 logger.info(f'Enter iteration {self.niter}.')
                 if self.timestep_adjuster is not None:
                     logger.info(f'Model timestep: {self.timestep}.')
-
+            
+            # Save the old radiative imbalance at the TOA. It is compared with
+            # values after adjustment to check if the model has converged.
+            
+            if self.niter != 0:
+             N = (
+                  (self.radiation["sw_flxd"][0,-1] +
+                   self.radiation["lw_flxd"][0,-1]
+                  ) -
+                  (self.radiation["sw_flxu"][0,-1] +
+                   self.radiation["lw_flxu"][0,-1]
+                  )
+                 )
+            else:
+             N = 0
```

Comment:
The radiation class provides a convenience variable `self.radiation["toa"]`, which should contain exactly this value.

Also, please indent blocks with [four spaces](https://www.python.org/dev/peps/pep-0008/#indentation).